battle-service/src/main/java/org/astashonok/battleservice/entities/Battle.java

```diff
@@ -0,0 +1,89 @@
+package org.astashonok.battleservice.entities;
+
+import com.vladmihalcea.hibernate.type.array.DoubleArrayType;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.astashonok.battleservice.constants.BattleConstants;
+import org.astashonok.battleservice.models.BattleStatus;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+
+import javax.persistence.*;
+import java.util.Arrays;
+import java.util.Objects;
+
+@Entity
+@Table(name = "battles")
+@TypeDef(name = "board_array", typeClass = DoubleArrayType.class)
+@Getter
+@Setter
+@NoArgsConstructor
+public class Battle {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private int boardHeight;
+    private int boardWidth;
+
+    @Type(type = "board_array", parameters = @org.hibernate.annotations.Parameter(name = "sql_array_type", value = "integer"))
+    @Column(columnDefinition = "integer[][]")
+    private int[][] board;
```

Comment:
do we really need to store board matrix?, or we can store list of moves only?
in this case you can avoid using nextMoveParticipantId but store(optionally lastMoveParticipantId)